skimage/exposure/exposure.py

```diff
@@ -348,6 +348,8 @@ def rescale_intensity(image, in_range='image', out_range='dtype'):
 
     image = np.clip(image, imin, imax)
 
+    if imin == imax:
+        return image
```

Comment:
I would really like it if the only thing that was avoided here was the division by zero.

The code path should go through the same trajectory multiplying by `(omax - omin)` and adding `omin`.

Sound riddiculous, but sometimes not going through that code path can hide nasty bugs due to returning a view of the same image, as opposed to a new array like the code below does.

Reply:
Ah I see. So should I do the following check instead of returning an original copy? 
```if imin` != imax: image = (image - imin) / float(imax - imin)```